components/sup/src/main.rs

```diff
@@ -156,6 +220,69 @@ fn sub_config(m: &ArgMatches) -> Result<()> {
     Ok(())
 }
 
+fn sub_load(m: &ArgMatches) -> Result<()> {
+    if m.is_present("VERBOSE") {
+        sup::output::set_verbose(true);
+    }
+    if m.is_present("NO_COLOR") {
+        sup::output::set_no_color(true);
+    }
+    let cfg = mgrcfg_from_matches(m)?;
+    let ident_or_artifact = m.value_of("PKG_IDENT_OR_ARTIFACT").unwrap();
+    let ident = if Path::new(ident_or_artifact).is_file() {
+        let env_or_default = henv::var(DEPOT_URL_ENVVAR).unwrap_or(DEFAULT_DEPOT_URL.to_string());
+        let depot_url = m.value_of("DEPOT_URL").unwrap_or(&env_or_default);
+        let ident = PackageArchive::new(ident_or_artifact).ident()?;
+        let mut ui = UI::default();
+        outputln!("Installing local artifact {}",
+                  Yellow.bold().paint(ident_or_artifact));
+        common::command::package::install::start(&mut ui,
+                                                 &depot_url,
+                                                 &ident.to_string(),
+                                                 PRODUCT,
+                                                 VERSION,
+                                                 Path::new(&*FS_ROOT_PATH),
+                                                 &fs::cache_artifact_path(None),
+                                                 false)?;
+        ident
+    } else {
+        PackageIdent::from_str(ident_or_artifact)?
+    };
+    let default_spec = ServiceSpec::default_for(ident);
+    let spec_file = Manager::spec_path_for(&cfg, &default_spec);
+    if let Ok(spec) = ServiceSpec::from_file(&spec_file) {
+        if !m.is_present("FORCE") {
+            return Err(sup_error!(Error::ServiceLoaded(spec.ident)));
+        }
+    }
+    let mut spec = spec_from_matches(default_spec.ident, m)?;
+    spec.start_style = StartStyle::Persistent;
+    Manager::save_spec_for(&cfg, spec)
```

Comment:
The non-arg parsing logic probably should go into a proper subcommand module/function. There isn't currently any user feedback from this subcommand and that messaging can be duplicated in `hab-sup start` by invoking the same logic.